src/EditorFeatures/Core/Implementation/ExtractInterface/AbstractExtractInterfaceCommandHandler.cs

```diff
@@ -47,59 +48,63 @@ public VSCommanding.CommandState GetCommandState(ExtractInterfaceCommandArgs arg
 
         public bool ExecuteCommand(ExtractInterfaceCommandArgs args, CommandExecutionContext context)
         {
-            var document = args.SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
-            if (document == null)
-            {
-                return false;
-            }
-
-            var workspace = document.Project.Solution.Workspace;
-
-            if (!workspace.CanApplyChange(ApplyChangesKind.AddDocument) ||
-                !workspace.CanApplyChange(ApplyChangesKind.ChangeDocument))
+            using (context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Extract_Interface))
             {
-                return false;
-            }
-
-            var supportsFeatureService = document.Project.Solution.Workspace.Services.GetService<IDocumentSupportsFeatureService>();
-            if (!supportsFeatureService.SupportsRefactorings(document))
-            {
-                return false;
-            }
+                var document = args.SubjectBuffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChangesAsync(
+                    context.OperationContext).WaitAndGetResult(context.OperationContext.UserCancellationToken);
+                if (document == null)
+                {
+                    return false;
+                }
+
+                var workspace = document.Project.Solution.Workspace;
+
+                if (!workspace.CanApplyChange(ApplyChangesKind.AddDocument) ||
+                    !workspace.CanApplyChange(ApplyChangesKind.ChangeDocument))
+                {
+                    return false;
+                }
```

Comment:
these shoudl be front loaded.  It's bad that we'll do a super expensive amount of work *showing the user a blocking dialog*, then we'll just bail out.